trusted_application_demo/optee_ta_security/src/ta_aes.c

```diff
@@ -0,0 +1,405 @@
+/*
+* Copyright (c) 2018, ARM Limited.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+
+#include "ta_public.h"
+#include "ta_rsa.h"
+#include "ta_aes.h"
+#include "ta_utils.h"
+
+static TEE_OperationHandle aes_encrypt_op_handle[TA_PUBLIC_MAX_KEY_STORAGE];
+static TEE_OperationHandle aes_decrypt_op_handle[TA_PUBLIC_MAX_KEY_STORAGE];
+static TEE_OperationHandle hmac_op_handle[TA_PUBLIC_MAX_KEY_STORAGE];
```

Comment:
As I know, it should support handle number which define by `TA_PUBLIC_MAX_KEY_STORAGE`, such as `aes_decrypt_op_handle[id]`.
But I find only first index is used actually, such as:
`result = ta_aes_hmac_digest(0, message, size_message,
               hmac_digest, size_hmac_digest);`
Why not let the CA to manage the operate handles?

Reply:
The application has been simplified for this example. At first, the goal of this was to have only one AES key per topic. It seems that I forgot to refactor this part of the code. Thanks.